server/routes/surveyRoutes.js

```diff
@@ -22,36 +22,43 @@ module.exports = app => {
     res.send('Thanks for voting!');
   });
 
-  app.post('/api/surveys/webhooks', (req, res) => {
-    const p = new Path('/api/surveys/:surveyId/:choice');
-
-    _.chain(req.body)
-      .map(({ email, url }) => {
-        const match = p.test(new URL(url).pathname);
-        if (match) {
-          return { email, surveyId: match.surveyId, choice: match.choice };
-        }
-      })
-      .compact()
-      .uniqBy('email', 'surveyId')
-      .each(({ surveyId, email, choice }) => {
-        Survey.updateOne(
-          {
-            _id: surveyId,
-            recipients: {
-              $elemMatch: { email: email, responded: false }
+  app.post('/api/surveys/webhooks', async (req, res) => {
+    try {
+      const p = new Path('/api/surveys/:surveyId/:choice');
+
+      const surveys = req.body
+          .map(surveyData => {
+            const match = p.test(new URL(surveyData.url).pathname);
+            if (match) {
+              return { email: surveyData.email, surveyId: match.surveyId, choice: match.choice };
+            }
+          })
+          .filter(item => item !== null && item !== undefined);
```

Comment:
First, I removed lodash as much as possible. I ended up using it in one place later for simplicity, but it's mostly not needed. Vanilla Javascript has built in functions to map and filter arrays, so I use those instead of lodash here.

This builds an array of survey objects, then filters out any null or undefined objects. Not sure if this is really needed, since I don't know why there weould be anything null in the list, but I haven't looked at the front-end.

Ultimately, this does the same thing as the lodash, but...without it since it isn't needed.